manta-pay/src/crypto/ecc/arkworks.rs

```diff
@@ -545,3 +539,80 @@ where
         )
     }
 }
+
+impl<C, CV> PointAdd<Compiler<C>> for GroupVar<C, CV>
+where
+    C: ProjectiveCurve,
+    CV: CurveVar<C, ConstraintField<C>>,
+{
+    type Output = Self;
+
+    fn add(&self, rhs: &Self, compiler: &mut Compiler<C>) -> Self::Output {
+        let _ = compiler;
+        let mut result = self.0.clone();
+        result += &rhs.0;
+        Self::new(result)
+    }
+}
+
+impl<C, CV> PointDouble<Compiler<C>> for GroupVar<C, CV>
+where
+    C: ProjectiveCurve,
+    CV: CurveVar<C, ConstraintField<C>>,
+{
+    type Output = Self;
+
+    fn double(&self, compiler: &mut Compiler<C>) -> Self::Output {
+        let _ = compiler;
+        Self::new(self.0.double().expect("Doubling is not allowed to fail."))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ark_ec::ProjectiveCurve;
+    use ark_ed_on_bls12_381::constraints::EdwardsVar;
+    use ark_ed_on_bls12_381::EdwardsProjective;
+    use manta_crypto::constraint::ConstraintSystem;
+    use manta_crypto::ecc::{PreprocessedScalarMulTable, ScalarMul};
+    use manta_crypto::rand::OsRng;
+    fn preprocessed_scalar_mul_test_template<C, CV, const N: usize>(rng: &mut OsRng)
```

Comment:
Please leave a newline between the imports and the first definition. Also, add a documentation string here.

Reply:
done